tests/utils/test_database.py

```diff
@@ -0,0 +1,109 @@
+from urllib.parse import urlparse
+
+import pytest
+from airflow.hooks.base import BaseHook
+from airflow.providers.sqlite.hooks.sqlite import SqliteHook
+from sqlalchemy import text
+from sqlalchemy.engine.base import Engine
+
+from astro.constants import Database
+from astro.utils.database import get_database_name, get_sqlalchemy_engine, run_sql
+from astro.utils.dependencies import BigQueryHook, PostgresHook, SnowflakeHook
+
+
+def describe_get_database_name():
+    def with_google_bigquery_hook():
+        hook = BigQueryHook()
+        db = get_database_name(hook)
+        assert db == Database.BIGQUERY
+        engine = hook.get_sqlalchemy_engine()
+        db = get_database_name(engine)
+        assert db == Database.BIGQUERY
+
+    def with_snowflake_hook():
+        hook = SnowflakeHook(snowflake_conn_id="snowflake_conn")
+        db = get_database_name(hook)
+        assert db == Database.SNOWFLAKE
+        engine = hook.get_sqlalchemy_engine()
+        db = get_database_name(engine)
+        assert db == Database.SNOWFLAKE
```

Comment:
Seems like these can be shortened with some functional magic, but maybe that would make things hard to read?

Maybe using behaves_like?

Something like:

```
def get_db_func(hook, db_name):
    db = get_database_name(hook)
    assert db == db_name
    engine = hook.get_sqlalchemy_engine()
    db = get_database_name(engine)
    assert db == db_name


def describe_get_database_name():
    @behaves_like(get_db_func)
    def with_snowflake_hook():
        @pytest.fixture
        def hook():
            return SnowflakeHook(snowflake_conn_id="snowflake_conn")

        @pytest.fixture
        def db_name():
            return Database.SNOWFLAKE
        
    @behaves_like(get_db_func)
    def with_postgres_hook():
        @pytest.fixture
        def hook():
            return PostgresHook()

        @pytest.fixture
        def db_name():
            return Database.POSTGRES
```

I realized it might not save code lines in this specific case, but it at least would help ensure that all the tests are run the same way

Reply:
I believe tests should be kept simple and with the least logic as possible - I'd rather keep the previous implementation.

Reply:
Ok sounds good

Reply:
Absolutely! This pattern can certainly simplify other test cases!